src/software/thunderscope/thunderscope_main.py

```diff
@@ -192,54 +198,74 @@
     #
-    # When we are running with real robots. We want to run 1 instance of AI
-    # and 1 instance of RobotCommunication which will send/recv packets over
-    # the provided multicast channel.
-    if args.run_blue:
-
-        tscope = Thunderscope(
-            layout_path=args.layout,
-            load_blue=True,
-            load_yellow=False,
-            load_diagnostics=True,
-            load_gamecontroller=False,
-            visualization_buffer_size=args.visualization_buffer_size,
-        )
-
-        proto_unix_io = tscope.blue_full_system_proto_unix_io
-        runtime_dir = args.blue_full_system_runtime_dir
-        friendly_colour_yellow = False
-        debug = args.debug_blue_full_system
-    elif args.run_yellow:
+    # When we are running with real robots. Possible Running Options:
+    #
+    #       Run Blue
+    #       Run Yellow
+    #       Run Blue + Diagnostics
+    #       Run Yellow + Diagnostics
+    #       Run Diagnostics
+    #
+    # We want to run either 1 instance of AI or 1 instance of RobotCommunication or both which will
+    # send/recv packets over the provided multicast channel.
 
+    if args.run_blue or args.run_yellow or args.run_diagnostics:
         tscope = Thunderscope(
             layout_path=args.layout,
-            load_blue=False,
-            load_yellow=True,
-            load_diagnostics=True,
+            load_blue=bool(args.run_blue),
+            load_yellow=bool(args.run_yellow),
+            load_diagnostics=bool(args.run_diagnostics),
             load_gamecontroller=False,
             visualization_buffer_size=args.visualization_buffer_size,
         )
 
-        proto_unix_io = tscope.yellow_full_system_proto_unix_io
-        runtime_dir = args.yellow_full_system_runtime_dir
-        friendly_colour_yellow = True
-        debug = args.debug_yellow_full_system
+        full_system_proto_unix_io = None
+        diagnostics_proto_unix_io = None
+
+        if args.run_blue:
+            full_system_proto_unix_io = tscope.blue_full_system_proto_unix_io
+            runtime_dir = args.blue_full_system_runtime_dir
+            friendly_colour_yellow = False
+            debug = args.debug_blue_full_system
+        elif args.run_yellow:
+            full_system_proto_unix_io = tscope.yellow_full_system_proto_unix_io
+            runtime_dir = args.yellow_full_system_runtime_dir
+            friendly_colour_yellow = True
+            debug = args.debug_yellow_full_system
+
+        if args.run_diagnostics:
+            diagnostics_proto_unix_io = tscope.robot_diagnostics_proto_unix_io
+
+        if args.run_blue or args.run_yellow:
+            with ProtoLogger(
+                args.blue_full_system_runtime_dir,
+            ) as blue_logger, ProtoLogger(
+                args.yellow_full_system_runtime_dir,
+            ) as yellow_logger, FullSystem(
+                runtime_dir, debug, friendly_colour_yellow
```

Comment:
I think we can get away with just having 1 `ProtoLogger` since we shouldn't be running 2 FullSystems at the same time during a game anyways